moabb/datasets/sosulski2019.py

```diff
@@ -0,0 +1,175 @@
+        code = "Spot Pilot P300 dataset"
+        interval = [-0.2, 1] if interval is None else interval
+        super().__init__(
+            subjects=list(range(1, 13 + 1)),
+            sessions_per_subject=1,
+            events=dict(Target=21, NonTarget=1),
+            code=code,
+            interval=interval,
+            paradigm="p300",
+            doi="10.6094/UNIFR/154576",
+        )
+
+    @staticmethod
+    def _map_subject_to_filenumber(subject_number):
+        # The ordering of the uploaded files on freidok makes no sense, this function maps subject_numbers to corresponding files
+        mapping = [5, 2, 4, 6, 3, 1, 10, 7, 12, 9, 8, 11, 13]
+        return mapping[subject_number - 1]
+
+    @staticmethod
+    def filename_trial_info_extraction(filepath):
+        info_pattern = "Oddball_Run_([0-9]+)_Trial_([0-9]+)_SOA_[0-9]\\.([0-9]+)\\.vhdr"
+        filename = filepath.split(os.path.sep)[-1]
+        trial_info = dict()
+        re_matches = re.match(info_pattern, filename)
+        trial_info["run"] = int(re_matches.group(1))
+        trial_info["trial"] = int(re_matches.group(2))
+        trial_info["soa"] = int(re_matches.group(3))
+        return trial_info
+
+    def _get_single_run_data(self, file_path):
+        non_scalp_channels = ["EOGvu", "x_EMGl", "x_GSR", "x_Respi", "x_Pulse", "x_Optic"]
+        raw = mne.io.read_raw_brainvision(
+            file_path, misc=non_scalp_channels, preload=True
+        )
+        raw.set_montage("standard_1020")
+        if self.reject_non_iid:
+            raw.set_annotations(raw.annotations[7:85])  # non-iid rejection
+        return raw
+
+    def _get_single_subject_data(self, subject):
+        """return data for a single subject"""
+
+        file_path_list = self.data_path(subject)
+        sessions = {}
+
+        for p_i, file_path in enumerate(file_path_list):
+            file_exp_info = Sosulski2019.filename_trial_info_extraction(file_path)
+            soa = file_exp_info["soa"]
+            # trial = file_exp_info["trial"]
+            if soa == 60 and not self.load_soa_60:
+                continue
+            if self.use_soas_as_sessions:
+                session_name = f"session_1_soa_{soa}"
+            else:
+                session_name = "session_1"
+
+            if session_name not in sessions.keys():
+                sessions[session_name] = {}
+
+            run_name = f"run_{p_i + 1}_soa_{p_i}"
+            sessions[session_name][run_name] = self._get_single_run_data(file_path)
+
+        return sessions
+
+    def data_path(
+        self, subject, path=None, force_update=False, update_path=None, verbose=None
+    ):
+
+        if subject not in self.subject_list:
+            raise (ValueError("Invalid subject number"))
+
+        # check if has the .zip
+        file_number = Sosulski2019._map_subject_to_filenumber(subject)
+        url = f"{SPOT_PILOT_P300_URL}/FILE{file_number}/content"
+        path_zip = dl.data_path(url, "spot")
```

Comment:
You should use `data_dl` instead of `data_path` as the later is deprecated.

Reply:
Hm, is there a functional difference in data_dl? I tried simply replacing the function name, but `data_dl` does not work if the data is already available. I.e., when the data needs to be downloaded the code works, but when it is already downloaded, the program is stuck.

Reply:
Oh nevermind, it only looks stuck because I amnot used to the progress bar yet... :D

Reply:
Yes, you need to install tqdm. The pyproject file is updated (`poetry install` should do the trick) but I need to regenerate the requirement file if you use a more ad hoc process.